### PR TITLE
ci: enable renovate bot

### PR DESCRIPTION
**Pull request by alan-agius4**
With this change dependencies are now managed via Renovate bot.

- Major updates are scheduled after 10pm and before 4am every weekday and unscheduled during weekends
- lock file maintenance, every Monday
- minor and patch updates are grouped and scheduled after 1am every Thursday

### CODE REVIEW

File: renovte.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  "separateMajorMinor": false,
+  "prHourlyLimit": 2,
+  "labels": ["comp: build", "dependencies"],
+  "timezone": "America/Tijuana",
+  "lockFileMaintenance": {
+    "enabled": true
+  },
+  "schedule": ["after 10pm every weekday", "before 4am every weekday", "every weekend"],
+  "baseBranches": ["main"],
+  "ignoreDeps": ["@types/node", "@types/selenium-webdriver", "selenium-webdriver"],
```
Comment (devversion) [inline renovte.json:14]: Is there context on these ignore packages?

**Review by devversion — APPROVED** [review]
LGTM. I personally never favorized automatic version updates as I believe updating dependencies manually encourages more thorough investigation of how the update would affect the code (especially for the Bazel stuff this is needed often).

File: renovte.json (hunk 1 above)
Comment (alan-agius4) [reply renovte.json:14]: -`selenium-webdriver` version 4 beta which is current marked as latest.
- `@types/node` seems to be pinned to the min supported version fo the Angular framework and hence the update needs to happen manually.

Comment (devversion) [reply renovte.json:14]: 👍 thanks. good to capture this here.

**Comment by alan-agius4** [comment]
> I personally never favorized automatic version updates as I believe updating dependencies manually encourages more thorough investigation of how the update would affect the code (especially for the Bazel stuff this is needed often).

I think in most cases this is only true for major versions. I personally feel that updating dependencies manually is too much of a burden. (Side note: `rules_nodejs` repo is also managed by Renovate).

In the CLI we have been using Renovate for a couple of years now and truly cannot imagine these updates to be done manually nowadays. That's just my personal opinion of course. If code coverage is low, I'd agree that it could be problematic.

**Comment by devversion** [comment]
yeah, it's definitely up to preference. We've been doing very fine without Renovate/Dependabot for years in the COMP repo (now it's enabled; but not a big fan still 😄; especially if you need to push to a Renovate PR to fix it up)

**Review by josephperrott — APPROVED** [review]

**Comment by alan-agius4** [comment]
@josephperrott note: you need to enable renovate once this is merged.

**Pull request closed without merging**